src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs

```diff
@@ -170,11 +171,18 @@ private static FileStream CreateSharedBackingObject(Interop.Sys.MemoryMappedProt
                 ?? CreateSharedBackingObjectUsingFile(protections, capacity, inheritability);
         }
 
-        private static FileStream? CreateSharedBackingObjectUsingMemory(
+        private static unsafe FileStream? CreateSharedBackingObjectUsingMemory(
            Interop.Sys.MemoryMappedProtections protections, long capacity, HandleInheritability inheritability)
         {
-            // The POSIX shared memory object name must begin with '/'.  After that we just want something short and unique.
-            string mapName = string.Create(null, stackalloc char[128], $"/corefx_map_{Guid.NewGuid():N}");
+            const int MaxSharedMemoryObjectNameLength = 32; // SHM_NAME_MAX on OSX ARM64, on other systems it's equal PATH_MAX (250)
+            // The POSIX shared memory object name must begin with '/'.  After that we just want something short (32) and unique.
+            string mapName = string.Create(MaxSharedMemoryObjectNameLength, 0, (span, state) =>
+            {
+                Guid.NewGuid().TryFormat(span, out int charsWritten, "N");
+                Debug.Assert(charsWritten == MaxSharedMemoryObjectNameLength);
+                span[0] = '/';
+                "dotnet".CopyTo(span.Slice(1));
```

Comment:
```suggestion
                "/dotnet".CopyTo(span);
```

Reply:
@jkotas do you mean getting the 16 bytes out of Guid and mapping it into a string using the base64 encoding?

I'll check on Monday whether all characters used by the base64 encoding are valid for map names

Reply:
You can use proper random number generator `RandomNumberGenerator.Fill` if you go with Base64 encoding.

Note that Guid contains less entropy than what you may think. From https://docs.microsoft.com/en-us/dotnet/api/system.guid.newguid?view=net-6.0 : The generated GUID contains 122 bits of strong entropy.

Reply:
@jkotas I've given your idea a try. It works, but:

* using `RandomNumberGenerator.Fill`, requires to add a new dependency to `System.Security.Cryptography`. Since I have to backport this fix to 6.0 I am hesitant of adding any new dependencies.
* I've [tested](https://github.com/dotnet/runtime/pull/64099/commits/64bb94b4ce1f37490940f245dfcda86b2c7f2683) and using an additional `/` in the name (produced by Base64) fails with EINVAL. Of course we can easily replace it.

I would prefer to keep using `Guid`, but also implement a simply retry logic for `EEXIST` (the error we get when a map with such name already exists). @jkotas what do you think?


<details>

```cs
const int MaxSharedMemoryObjectNameLength = 32; // SHM_NAME_MAX on OSX ARM64, on other systems it's equal PATH_MAX (250)
// The POSIX shared memory object name must begin with '/'.  After that we just want something short (32) and unique.
string mapName = string.Create(MaxSharedMemoryObjectNameLength, 0, (span, state) =>
{
    const string Prefix = "/dotnet_map_";
    Prefix.CopyTo(span);
    const int ByteCount = 15;
    Debug.Assert(ByteCount == (MaxSharedMemoryObjectNameLength - Prefix.Length) / 4 * 3); // Base64 turns every 3 bytes into 4 characters
    Span<byte> buffer = stackalloc byte[ByteCount];
    RandomNumberGenerator.Fill(buffer);
    Convert.TryToBase64Chars(buffer, span.Slice(Prefix.Length), out int charsWritten);
    Debug.Assert(charsWritten == MaxSharedMemoryObjectNameLength - Prefix.Length);

    for (int i = Prefix.Length; i < span.Length; i++)
    {
        if (span[i] == '/' || span[i] == '+') // used by Base64 encoding, but invalid for shm_open (EINVAL)
        {
            span[i] = '_';
        }
    }
});
```

</details>

Reply:
Ok, keeping Guid sounds good to me.